lib/Driver/ToolChains.cpp

```diff
@@ -1211,12 +1211,33 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(LibProfile));
   }
 
-  // FIXME: We probably shouldn't be adding an rpath here unless we know ahead
-  // of time the standard library won't be copied.
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back("-rpath");
-  Arguments.push_back("-Xlinker");
-  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+  if (getTriple().isAndroid()) {
+    // FIXME: These should be set in CMake.
+    Arguments.push_back("-target");
+    Arguments.push_back("armv7-none-linux-androideabi");
+
+    const char* ndkhome = getenv("ANDROID_NDK_HOME");
```

Comment:
This is customary with respect to the Android ndk build scripts. You set this and the build scripts handle constructing the paths given to gcc/clang while compiling. To my knowledge gcc/clang don't handle this themselves in any way.